crates/subspace-wasm-tools/src/lib.rs

```diff
@@ -47,40 +46,51 @@ pub fn create_runtime_bundle_inclusion_file(
                 out_components.next();
             }
 
-            if let Component::Normal(profile) = out_components
-                .next()
-                .expect("OUT_DIR has CARGO_TARGET_DIR as prefix")
-            {
-                profile
-                    .to_str()
-                    .expect("Profile is always a utf-8 string")
-                    .to_string()
-            } else {
-                unreachable!("OUT_DIR has CARGO_TARGET_DIR as prefix")
-            }
+            let mut path = target_dir;
+
+            let profile = loop {
+                if let Component::Normal(maybe_profile) = out_components
+                    .next()
+                    .expect("OUT_DIR has CARGO_TARGET_DIR as prefix")
+                {
+                    let maybe_profile = maybe_profile
+                        .to_str()
+                        .expect("Profile is always a utf-8 string")
+                        .to_string();
```

Comment:
`to_string()` is not necessary.

Reply:
It actually is since in one case we compare two strings and in other we return it below, so I think it has to be owned.

Reply:
Still compiles for me without `to_string()`.